### ISSUE TITLE
AttributeError in simplify_network when calling aggregategenerators()

### ISSUE THREAD

**Issue by ekatef** [pz-max/pypsa-africa#558]
## Checklist

- [x] I am using the current [`main`](https://github.com/pypsa-meets-earth/pypsa-earth/tree/main) branch or the latest [release](https://github.com/pypsa-meets-earth/pypsa-earth/releases). Please indicate.
- [x] I am running on an up-to-date [`pypsa-earth` environment](https://github.com/pypsa-meets-earth/pypsa-earth/blob/main/envs/environment.yaml). Update via `conda env update -f envs/environment.yaml`.

## Describe the Bug

When running the workflow for Nigeria, an error appears in `simplify_network` in a call of `aggregategenerators()`. CI however seems to work properly.

## Error Message

```bash
Traceback (most recent call last):
  File "~/pypsa-earth/.snakemake/scripts/tmp8s5o3g1n.simplify_network.py", line 538, in <module>
    n, stub_map = remove_stubs(
  File "~/pypsa-earth/.snakemake/scripts/tmp8s5o3g1n.simplify_network.py", line 447, in remove_stubs
    _aggregate_and_move_components(
  File "~/pypsa-earth/.snakemake/scripts/tmp8s5o3g1n.simplify_network.py", line 256, in _aggregate_and_move_components
    generators, generators_pnl = aggregategenerators(
  File "/Users/ekaterina/opt/miniconda3/envs/pypsa-earth-upd/lib/python3.10/site-packages/pypsa/networkclustering.py", line 82, in aggregategenerators
    weighting = generators.weight.groupby(grouper, axis=0).transform(normed_or_uniform)
  File "/Users/ekaterina/opt/miniconda3/envs/pypsa-earth-upd/lib/python3.10/site-packages/pandas/core/generic.py", line 5902, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'weight'
```

**Comment by ekatef** [comment]
The issue seems to be region-related: e.g. it appears for Nigeria but doesn't for Kazakhstan

**Comment by ekatef** [comment]
Have checked that the type conversion fix is fetched properly from #553 but the problem still persists
@davide-f, do you have any ideas on what can happen here?

**Comment by ekatef** [comment]
Update: this issue is caused by using a wrong cutout. 

Probably, it's worth to add a cutout check to the bug report template. Adding a cutout check to the workflow is still contained as a part of #528
